### PR TITLE
Migrated trace data to profiler library

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
Closes [#1602](https://github.com/foundry-rs/starknet-foundry/issues/1602)

## Introduced changes

- Migrates to profiler trace data struct

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Comment by MaksymilianDemitraszek** [comment]
Can be reviewed already, but before merging:

- [ ] https://github.com/software-mansion/cairo-profiler/pull/34/has to be merged
- [ ] git dependency has to be updated to point relevant commit

**Review by piotmag769 — COMMENTED** [review]
We should prob drop the starknet.rs dependency from cairo-profiler first as you suggested

**Comment by MaksymilianDemitraszek** [comment]
`starknet.rs`? you mean `starknet_api`?

File: crates/forge-runner/src/trace_data.rs [hunk 1]
```diff
@@ -26,310 +24,198 @@ pub const TRACE_DIR: &str = ".snfoundry_trace";
...
+        class_hash,
+        code_address,
+        entry_point_type,
+        entry_point_selector,
+        calldata,
+        storage_address,
+        caller_address,
+        call_type,
+        initial_gas: _,
+    } = value;
```
Comment (war-in) [inline crates/forge-runner/src/trace_data.rs:117]:
```suggestion
    let CallEntryPoint {
        class_hash,
        code_address,
        entry_point_type,
        entry_point_selector,
        calldata,
        storage_address,
        caller_address,
        call_type,
        ..
    } = value;
```

?

File: crates/forge-runner/src/trace_data.rs [hunk 2]
```diff
@@ -26,310 +24,198 @@ pub const TRACE_DIR: &str = ".snfoundry_trace";
...
-                ProfilerDeprecatedSyscallSelector::StorageRead
-            }
-            DeprecatedSyscallSelector::StorageWrite => {
-                ProfilerDeprecatedSyscallSelector::StorageWrite
-            }
-        }
+fn build_profiler_call_type(value: CallType) -> ProfilerCallType {
+    match value {
+        CallType::Call => ProfilerCallType::Call,
+        CallType::Delegate => ProfilerCallType::Delegate,
```
Comment (war-in) [inline crates/forge-runner/src/trace_data.rs:218]: What do you think about implementing `From` `CallType` trait for`ProfilerCalltype`?

**Review by war-in — APPROVED** [review]

File: crates/forge-runner/src/trace_data.rs (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/trace_data.rs:218]: It would require profiler to depend on blockifier, which we want to avoid.

Comment (war-in) [reply crates/forge-runner/src/trace_data.rs:218]: Right, resolved then

File: crates/forge-runner/src/trace_data.rs [hunk 3]
```diff
@@ -26,310 +27,159 @@ pub const TRACE_DIR: &str = ".snfoundry_trace";
...
+        contract_name = Some(String::from(TEST_CODE_CONTRACT_NAME));
+        function_name = Some(String::from(TEST_CODE_FUNCTION_NAME));
+    }
+    ProfilerCallEntryPoint {
+        entry_point_type: build_profiler_entry_point_type(entry_point_type),
+        entry_point_selector: EntryPointSelector(format!("{}", entry_point_selector.0)),
+        contract_address: ContractAddress(format!(
+            "0x{}",
+            hex::encode(storage_address.0.key().bytes())
+        )),
```
Comment (piotmag769) [inline crates/forge-runner/src/trace_data.rs:97]:
```suggestion
        contract_address: ContractAddress(format!(
            "{}",
            storage_address.0.key()
        )),
```

and we can remove `hex` dependency

Comment (MaksymilianDemitraszek) [reply crates/forge-runner/src/trace_data.rs:97]: Good point, just blindly copied it from the profiler code.

**Pull request merged**